app/Filament/Resources/GameResource.php

```diff
@@ -322,6 +322,26 @@ public static function form(Form $form): Form
                             ->default('day')
                             ->reactive(),
                     ]),
+
+                Forms\Components\Section::make('Media')
+                    ->icon('heroicon-o-photo')
+                    ->description("Provide the game icon, screenshots, and box art.")
+                    ->columns(['md' => 2, 'xl' => 3, '2xl' => 4])
+                    ->schema([
+                        Forms\Components\FileUpload::make('ImageIcon')
+                            ->label('Game Icon')
+                            ->disk('local')
+                            ->directory('temp')
+                            ->visibility('private')
+                            ->image()
+                            ->acceptedFileTypes(['image/jpeg', 'image/png', 'image/gif'])
+                            ->maxSize(1024)
+                            ->rules('dimensions:width=96,height=96')
```

Comment:
Where are "uploaded" files stored? Do they automatically get cleaned up? I attached a wrongly sized file and it still appears to have been uploaded:
![image](https://github.com/user-attachments/assets/fc0eb787-8790-4520-9949-fe15274fc318)

Also, the error message remains after uploading a different (correctly sized) image.

And uploading that generated an error:
![image](https://github.com/user-attachments/assets/4b370c51-ec3e-408c-b5d8-7ba80c667c74)

What permission do I need?
If the user doesn't have permission, should the field even be there?

Reply:
"Draft" files (for lack of a better term) are uploaded to _storage/app/temp_. Filament only tries to associate them with the game when the user clicks the form submit button.

They are indeed automatically cleaned up (see `ProcessUploadedImageAction::cleanUpTempFiles()`), but it's a side effect. Maybe it would make sense for the cleanup code to run on some kind of schedule?

The permission needed is `GamePolicy::update()`.

Visibility is indeed conditional based on this policy in _GameResource.php_ on line 347.

Authorization passes is the user has any role of GAME_HASH_MANAGER, DEVELOPER_STAFF, DEVELOPER, or DEVELOPER_JUNIOR.

The authorization call occurs in `GameResource/Pages/Edit.php` on line 20.

My user account has the `DEVELOPER` role. This is what I see when I use the control locally:

https://github.com/user-attachments/assets/0c13702e-729d-482c-a13c-e5e1c5c6c748

Reply:
> "Draft" files (for lack of a better term) are uploaded to storage/app/temp.

It seems like it maintains the filename of the uploaded file. If two users both upload a "icon.png" at roughly the same time, will one overwrite the other?

Wrongly sized files don't appear in that folder.

Reply:
I'm still getting the error. And I have at least two of the roles you've identified.

https://github.com/user-attachments/assets/e45b35bd-e143-4567-a1d2-fa7d42d20661

Reply:
And the error still stays on screen when uploading a second (correctly sized) image.
![image](https://github.com/user-attachments/assets/abe34a10-1ee3-403c-986a-5fe0ca55d681)

You can see it in your video at the 25 second mark.